elixir - general, solution, two-fer remarks (#1344)
exercism/website-copy: automated-comments/elixir/two-fer/use_default_param.md

<!--
This error is raised when the solution doesn't use default parameters
-->

Use a [default parameter](https://elixir-lang.org/getting-started/modules-and-functions.html#default-arguments) for this solution.

> - [Default parameters](https://elixir-lang.org/getting-started/modules-and-functions.html#default-arguments) allows you to avoid writing duplicate function clauses for functions with different [arities](https://en.wikipedia.org/wiki/Arity).

Reviewer: ```suggestion
- [Default parameters](https://elixir-lang.org/getting-started/modules-and-functions.html#default-arguments) removes the need of writing duplicate function clauses for functions with different [arities](https://en.wikipedia.org/wiki/Arity).
```

Example how "you" could be removed.

```elixir
# Consider this function:
def foo(bar \\ "baz") do
  IO.puts(bar)
end
```

- This function now defines both `foo/0` and also `foo/1`

> - if `foo()` is called, `"baz"` is put to the stdio

Reviewer: ```suggestion
- if `foo()` is called, `"baz"` is put to the `stdio`
```

Reviewer 2: I'd prefer `stdout` without an article.

Author: Do you have an article which explains what the `group leader` is?

Reviewer 2: There is [`:erlang.group_leader/0`](http://erlang.org/doc/man/erlang.html#group_leader-0) as a function, whichs documentation makes a bad job at explaining it ;D

Also there is a very brief explanation in a section about the [`application` behaviour in Learn you some Erlang](https://learnyousomeerlang.com/building-otp-applications#the-application-behaviour).

I'm not aware of anything beyond that.

Author: Would it be fair to say 

```markdown
> When Elixir is compiled and run, the BEAM VM starts a process known as the `group leader` which represents standard input and output, this process is shared with all other processes to abstract input and output operations

For your reference:
- [Elixir-Lang - Getting Started - IO and the File System](https://elixir-lang.org/getting-started/io-and-the-file-system.html#processes-and-group-leaders)
- [Learn you some Erlang for great good - The application behavior](https://learnyousomeerlang.com/building-otp-applications#the-application-behaviour)

> - if `foo("bar")` is called, `"bar"` is put to the stdio

Reviewer: ```suggestion
- if `foo("bar")` is called, `"bar"` is put to the `stdio`
```

Reviewer 2: Similar as above:

```suggestion
- if `foo("bar")` is called, `bar` is written to the process leader
```

Author: I think I have resolved this in an update, see above review for comments
